---
title: "A PRIORI DECIDED PROCEDURE for 'ruff sperm' project"
author: "Martin Bulla, Wolfgang Forstmeier, Katrin Martin, Kim Teltscher, ?Jasmine L Loveland?, Clemens Küpper, Michael Lierz, Tomas Albrecht, David B Lank & Bart Kempenaers"
date: "`r Sys.time()`"
output: 
    html_document:
        toc: true
        toc_float: true
        toc_depth: 5
        code_folding: hide
bibliography: ruff_sperm.xml
link-citations: yes
---
<style type="text/css">

td {  /* Table  */
  font-size: 12px;
}
h1.title {
  font-size: 32px;
}
h1 { /* Header 1 */
  font-size: 24px;
}
h2 { /* Header 2 */
    font-size: 20px;
    
}
h3 { /* Header 3 */
  font-size: 16px;

}
h4 { /* Header 4 */
  font-size: 16px;
}
code.r{ /* Code block */
    font-size: 12px;
}
pre { /* Code block - determines code spacing between lines */
    font-size: 10px;
}
</style>

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

***

Code to load tools and prepare data:  
```{r tools, results="hide", warning = FALSE, message = FALSE}
  require(here)
  source(here::here('R/tools.R'))
  colors <- c("#999999", "#E69F00", "#56B4E9") #viridis(3)

  v = data.table(read_excel(here::here('Data/ruff_sperm_Vancouver_2018.xlsx'), sheet = 1))
  v = v[!is.na(sample_ID)]
  x = v[!duplicated(bird_ID)]
  
  s = data.table(read_excel(here::here('Data/ruff_males_Seewiesen.xlsx'), sheet = 1))
  setnames(s, 'Morph', 'morph') 
  sv = x[bird_ID%in%s$Ind_ID]

```

***

# Background 
Ruff *Calidris pugnax* is a lekking shorebird with three strikingly different mating morphs: aggressive ‘independents’, semi-cooperative ‘satellites’ and female-mimic ‘faeders’ [@vanRhijn1991; @Widemo1998; @Jukema2006]. 'Independents' (80 to 95% of the population) are dominant holders of display sites. Satellites (5 to 20%) do not defend display sites, but their presence on the lek assists with female attraction and allows them to 'steal' copulations. 'Faeders' (<1%) mimic females in their plumage and smaller size and also 'steal' copulations. The major differences in body and testis size, ornamentation, and mating behaviours of 'satellites' and 'faeders' develop via an autosomal inversion [@Kupper2015; @Lamichhaney2015]. Whether the morphs also differ in sperm morphology and velocity has not been investigated. 


# Research questions 
1. Do the three ruff morphs differ in the sperm morphology and velocity, as well as in the within- and between-individual variability of the sperm traits?  - see figure 1 [here](https://www.nature.com/articles/s41559-017-0236-1?platform=hootsuite)
2. Does sperm morphology predict sperm swimming speed? - see figure 2 [here](https://www.nature.com/articles/s41559-017-0236-1?platform=hootsuite)

# Predictions 

We expect no differences between the morphs in sperm morphology and velocity because the morphs differ little in spermatogenesis-related gene expression in testis [@Loveland2021].

```{r Predictions,  warning = FALSE, message = FALSE, fig.align="center", fig.width=3, fig.height=3, out.extra='style="float:right; padding:10px"'}
  set.seed(1)
  d = data.table(morph = factor(c( rep('independent', 60),
                                   rep('satelite', 30),
                                   rep('faeder', 15)
                                   ), 
                                levels = c('independent','satelite','faeder')
                                ),
                sperm_trait =  c(rnorm(60, mean = 1, sd = 0.5), 
                                 rnorm(30, mean = 2, sd = 0.5),
                                 rnorm(15, mean = 3, sd = 0.5) 
                                 ) 
                )
  #dev.new(width = 3.5, height = 3.5)
  ggplot(d, aes(x = morph, y = sperm_trait, col = morph)) +  
    geom_jitter() + theme_MB +  theme(legend.position="none") + xlab('Morph') + ylab('Sperm trait') +
    scale_colour_manual(values = colors)
 
```
On the other hand, the three morphs may differ in sperm traits because sperm length and swimming speed as well as investment in sperm numbers (testes size) is generally related to the intensity of sperm competition [CIT]. Indeed, 'independents' have relatively smaller testes then 'faeders' and 'satelites' [@Kupper2015]. In contrasts, dominant 'independents' on the lek obtain the highest number of copulations, the likelihood of copulation for other ‘independents’ and ‘satellites’ is driven by the amount of time an individual spends on the lek [@Vervoort2019].  Investing in pre-copulatory male-male competition may trade-off with investing in post-copulatory male-male competition. We thus predict that 'faeders' have the longest sperm, midpiece and tail and have sperm that is fastest of the three morphs (Figure 1 & 2). As faders invest little in pre-copulatory male-male competintion, we also expect 'faeders' sperm to differ the most from the sperm of 'independents', while we expect 'satelites', who partly invest in pre-copulatory male-male competition , but  to lesser extant than 'independents' [@Vervoort2019], to have intermediate sperm between the two (Figure 1). Consequently, we predict the smallest coefficient of variation in the sperm morphometry and velocity in 'faeders' as within- and between-individual variability in sperm traits is lower in species with higher sperm competition (@Kleven2008; @Lifjeld2010). Such optimizations shall increase 'faeders' chances of inseminating a female despite 'faeders' only sporadic chances to copulate [**CIT NEEDED or DELETE**]. 

**Figure 1** | Predicted relationship between morph type and sperm trait.

```{r Predictions2,  warning = FALSE, message = FALSE, fig.align="center", fig.width=4, fig.height=4}
  set.seed(1)
  d = data.table(morph = c( rep('independent', 60),
                                   rep('satelite', 30),
                                   rep('faeder', 15)
                                   ),
                midpiece =  c(rnorm(60, mean = 2, sd = 0.5), 
                                 rnorm(30, mean = 3, sd = 0.5),
                                 rnorm(15, mean = 4, sd = 0.5) 
                                 ),
                
                tail =  c(rnorm(60, mean = 2, sd = 0.5), 
                                 rnorm(30, mean = 3, sd = 0.5),
                                 rnorm(15, mean = 4, sd = 0.5) 
                                 ),
                velocity =  c(rnorm(60, mean = 2, sd = 0.5), 
                                 rnorm(30, mean = 3, sd = 0.5),
                                 rnorm(15, mean = 4, sd = 0.5) 
                                 )
                )
  d[,morph123 :=ifelse(morph == 'independent', 1, ifelse(morph == 'satelite', 2,3))]             

  colors_ <- colors[d$morph123]
  par(las = 1, cex.axis = 0.6, cex.lab = 0.8, cex.main = 0.8)
  s3d=scatterplot3d(d$midpiece, d$tail, d$velocity, pch = 16, type="h", 
              color=colors_, grid=TRUE, box=FALSE,
              xlab = "",
              ylab = "",
              zlab = "",
              x.ticklabs=c("short","","","","","long"),
              y.ticklabs=c("short","","","","long",""),
              z.ticklabs=c("slow","","","","","fast"),
              mar = c(3, 2, 0, 1.5)
              )     
  text(x = 7.5, y = 1, "Tail", srt = 0, cex = 0.8)
  text(x = 2.5, y = -0.5, "Mipiece", srt = 0,xpd = TRUE, cex = 0.8)
  text(x = -0.5, y = 2.5, "Velocity", srt = 90,xpd = TRUE, cex = 0.8)
  legend("bottom", legend = levels(factor(d$morph,levels = c('independent','satelite','faeder'))),
      col =  colors, pch = 16,xpd = TRUE, horiz = TRUE,inset = -0.125, bty = "n", cex = 0.7)
 
```
**Figure 2** | Predicted relationship between morph type and sperm traits.

# Methods #
## Individuals 
In 2018, we collected and stored in ~5% formalin the vas deferens from 15 sacrificed males (5 per morph) from the **Simon Fraser** University **colony**. The colony was founded with 110 ruffs hatched from wild eggs collected in Finland in 1985, 1989 and 1990 plus two 'faeder' males from the Netherlands in 2006 [@Lank1995; @Lank2013]. 

In 2021, we plan to collect sperm of all males from the **Max Planck colony** (`r nrow(s[morph == 'I'])` 'independents', `r nrow(s[morph == 'S'])` 'satelites', `r nrow(s[morph == 'F'])` 'faeders'; of which `r nrow(sv)` males were sampled already in Vancouver: `r nrow(sv[morph == 'r'])`  'independents', `r nrow(sv[morph == 's'])` 'satelites' and `r nrow(sv[morph == 'f'])` 'faeders'). The colony was founded in 2018 from the Simon Fraser colony. 

## Sperm sampling
Compared to passerines, shorebirds do not have a cloaca protuberance. Thus, abdominal massage and cloacal lavage method of sperm collection (e.g. @Knief2017; see the [specific protocol](https://raw.githack.com/MartinBulla/ruff_sperm/main/Protocols/protocol_sperm.html)) often lead to unclean samples contaminated by excrements. Such samples are then unsuitable for measuring sperm velocity. Hence, **in 2021** along with abdominal massage and cloacal lavage, we plan to collect sperm by electro‐stimulation [@Lierz2013]. The length and diameter of the electro-stimulation probe, as well as the electric current and the number of electric impulses will be adapted to the sampled individuals. The electro-stimulation should result in clean sperm samples suitable for velocity measurements. 

Sperm will be pipetted from the cloaca (~0.5–3μl) and immediately diluted in a preheated (40°C) Dulbecco’s Modified Eagle’s Medium (Advanced D-MEM, Invitrogen, USA). For the velocity measurements, an aliquot will be pipetted onto a standard 20μm two-chamber count slide (Leja, The Netherlands) placed on a  heating table kept at 40 °C. For the morphology measurements, an aliquot will be pipetted onto a microscopy slide and the rest of the sperm sample will be fixed in 100μl ~5% formalin solution.

## Sperm morphometry
From each sperm sample, we will pipet ~10μl onto a microscopy slide, let it dry at room temperature, and - in case of formalin-fixed samples - wash the slide gently under  distilled water to wash away the dried formalin and phosphate-buffered saline solution. Sperm from th vas deferens will be first extracted by cutting a piece from the middle of the tubules, plucking apart out the tissue with curved extraction forceps to release the sperm, and diluting the tissue the sperm with in athe phosphate phosphate-buffered saline solution.. 

We will inspect the slides with a light microscope ('Zeiss Axio Imager.M2') under 200x magnification.  To aid distinction of sperm parts, dried slides will be stained with Hoechst 33342 and Mitotracker Green FM  (not earlier than 48h) before photographing. For step by step slide preparation protocol see [here](https://raw.githack.com/MartinBulla/ruff_sperm/main/Protocols/protocol_sperm.html).

```{r Pic1, echo=FALSE, fig.cap="Picture 1 | Ruff sperm blue-stained nucleus by Hoechst 33342 and green-stained midpiece by Mitotracker Green FM", echo=FALSE}
knitr::include_graphics(here::here("Pictures/Pic1_staining_example.jpg"))
```
For each sample, we will photograph at least 10 intact normal-looking spermatozoa, with a 12 megapixel (4250 × 2838) digital camera ('Zeiss Axiocam 512 color' with pixel size of 3.1μm × 3.1μm). Note that 10 sperm per male were previously used to investigate the relationship between coefficient of variation in sperm trait and sperm competition (@Kleven2008; @Lifjeld2010). For each sperm, we will measure the length of the acrosome, the nucleus, the midpiece and the tail to the nearest 0.1μm using the open source software [ImageJ](https://imagej.nih.gov/ij/) (for a detailed sperm measuring protocol see [here](https://www.dropbox.com/s/ouqfqih4gzx6rq5/Protocol_imageJ_sperm_measuring_manual.docx?dl=0)). To minimize observer error, all measurements will be taken by one person (KT), who will be blind to the individual morph and will measure individual sperm from all morphs and sample in a random order (NOTA THAT THIS IS NOT THE CASE FOR THE VAS DEFERENS).

We will calculate total sperm length as the sum of all parts, head length as the sum of acrosome and nucleus length, and flagellum length as the sum of midpiece and tail length. We will compute coefficients of variation for each trait (CV = [SD/mean]) both within males and between males of each morph. If we unable to measure 10 sperm for each male, we will correct the coefficients of variation for variation in sample size (CVadj = [1 + 1/(4n)]*CV; [@Sokal1981]). 

Each measured sperm and sperm part will be numbered and referenced in the database (see [sperm measuring protocol](https://www.dropbox.com/s/ouqfqih4gzx6rq5/Protocol_imageJ_sperm_measuring_manual.docx?dl=0) and [database](https://www.dropbox.com/s/9uk7bcxmd4o1prx/ruff_sperm_morphology.xlsx?dl=0)). This ensures transparency and allows re-measurement of the same sperm by the same or different person.

## Sperm velocity
For each sperm sample we will record sperm velocity for approximately 45s in eight different fields of the slide under a 100x magnification with a digital camera (UI-1540-C, Olympus) mounted on a microscope (CX41, Olympus) fitted with a heating table (**?MODEL Tomas?**) kept at a constant temperature of 40°C. The videos will have **?SO and SO Tomas?** resolution. A single person will analyze each recorded field using the CEROS computer-assisted sperm analysis (CASA) system (Hamilton Thorne Inc., Beverly, Massa- chusetts, USA), visually inspect the tracked objects and exclude non-sperm objects and static spermatozoa from the analysis [@Laskemoen2010; @Cramer2016; @Opatová2016]. 

The dilution medium does not contain any spermatozoa attractants to guide the spermatozoa towards one direction; thus, we will use  curvilinear velocity rather than straight-line velocity as our measurement of sperm swimming speed [@Laskemoen2010]. We will report how many sperm cells per sperm sample will be recorded (median, 95%CI, range) and - if necessary - control for the (log-transformed) number of measured sperm cells per sample. 

## Inbreeding 
Because our sperm samples came and will come from males bred in captivity, we expect higher levels of inbreeding compared with males from wild populations. In birds and mammals (but not in insects) inbred males have a higher proportion of abnormal sperm and lower sperm velocity than outbred males [@Gomendio2000; @Heber2013; @Ala-Honkola2013; @Opatová2016]. Thus, our sperm velocity measurements may not reflect sperm velocity of wild ruffs. However, there is no evidence that the for morphology  of normal‐looking sperm (e.g., length, coefficient of variation) differs between inbred and outbred males [@Mehlis2012; @Ala-Honkola2013; @Opatová2016]. Based on these studies, we assume that our measurements reflect the variation in sperm morphology observed in wild ruffs. Nevertheless, we know the relatedness of individuals within our populations (pedigree) and hence we will investigate whether variation in inbreeding and genetic diversity is linked to the sperm traits and if so we will attempt to control for it in our models.

## Analysis plan 
Sample sizes will reflect the maximum available data. No data selection will be done conditional on the outcome of statistical tests. To avoid unconscious data dragging, we will first analyze the data on sperm traits blind to the males morphs. Specifically, we will first randomly assign male morph to the measurements, explore such data and finalize the analyze using such data. Only then we will run the analyses with true morphs. We will report all results, all data exclusions, all manipulations and all measures in the study at the [GitHub repository](https://github.com/MartinBulla/ruff_sperm). 

## Statistical analyses
All analyses will be in R [@R-Core-Team2020] using lmer() and glmer() functions of the lme4 package to fit linear and generalized linear mixed-effects models, and - if necessary - one of the package for fitting the pedigree structure as a random effect (e.g. [brms](https://mikheyev.github.io/brms-wam/), MCMCglmm, pedigreeMM). We will estimate the variance explained by the fixed effects of our mixed-effects models as marginal R2-values, using the r.squaredGLMM() function of the MuMIn package (v1.15.6). Model fits will be visually validated (by qq-plots of residuals and plots of residuals against fitted values). 

In general, we will fit two sets of linear models. A first set of models will have as a response variable th raw sperm trait measurements and will bee controlled for multiple measurements per male by fitting male identity as a random effect. A second set of models will have average sperm trait per male as a response and will be controlled for the number of sampled sperm per individual (**What is the best way of doing this? Wolfgang do you recommend weights?**). MB THE FIRST SET OF MODLES IS RATHER COMPLEX (if pedigree is used), THAT IS WHY I PROPOSE ALSO THE SIMPLER AVERAGE VARIANT. THE AVERAGE VARIANT WILL ALSO CORRESPONS WITH MODELS FOR VELOCITY. HOWEVER, IF THIS MAKES THINGS TO COMPLICATED, WE CAN RUN JUST THE FIRST SET.

To investigate whether sperm traits differ between the morphs, we will fit a model for each sperm trait separately, with the male morph (three-level factor) as an explanatory variable and - if necessary - the pedigree structure as a random effect.

To investigate whether sperm morphology explains variation in sperm velocity, we will first fit a linear model with average velocity per male as the dependent variable, and with total sperm length, head (*acrosome + nucleus*), midpiece and tail length (**?or rather flagelum - midpiece+tail?**), as well as their squared terms (after mean-centering) and all two-way interactions between the three linear terms, as explanatory variables. If necessary, we will control for a pedigree structure. In case, some of the sperm traits substantially correlate (Pearson's r>0.6), we will use only one of the correlated traits. We will then investigate whether predicted sperm velocity corresponds with observed sperm velocities. 

 <span style="color:grey">**THE FOLLOWING IS LIKELY NOT NEEDED:** To evaluate multicollinearity between all main effect predictors, we will estimate their variance inflation factor using the corvif() function [@Zuur2009] in R. A general guideline is that a variance inflation factor larger than 5 or 10 is large, indicating that the model has problems estimating the coefficient. However, this in general does not degrade the quality of predictions. If the VIF is larger than 1/(1-R2), where R2 is the Multiple R-squared of the regression, then that predictor is more related to the other predictors than it is to the response. Thus, if VIF is larger than 5 we can use the model for predictions.</span>

# Proposed time line
- ongoing - measuring sperm morphology from vas deferens samples
- May - data collection
- June - July - photographing and measuring sperm samples, quantifying sperm velocity
- August - September - analyses and 1st draft

# TO DECIDE
- Is it possible to add some of Clemens birds, e.g. faeders?
- Does it matter whether we let the microscope slides dry at room temperature or whether we heat fix them?
- Shall we use average sperm measurements in all analyses, given that for velocity-morphology relationship we will use average values?
- What sperm traits to use? Total length with/without acrosome, acrosome, head (acrosome + nucleus), midpiece, tail, flagelum (midpiece + tail). BART: total length, head, midpiece and tail
- Shall we investigate whether the sperm measurements are representative of each male and morph by calculating the repeatability of sperm measurements per male and morph, which will be obtained through 1,000 parametric bootstrap iterations [@Stoffel2017].
- Shall we use discriminant function analyses on the two most different sperm traits to predict the sperm morph? BART: Perhaps use all sperm traits, calculate PCA and plot PC1 vs PC2 with morph in different colours? Not sure this is necessary, but might be a nice way to illustrate the differences (if any)?
- Do we need to control for the pedigree, if residuals of the models are randomly distributed, i.e. there is no pedigree structure in residuals?
- Which packages to use for modeling: [brms](https://mikheyev.github.io/brms-wam/), MCMCglmm, pedigreeMM
- What is the sound way to control for number of measured sperm per individual (in case this is variable) - weights? (**Wolfgang?**) 
- When recording sperm velocity, shall we also make an additional record in a cellulose solution, which "mimics" conditions within the female reproductive tract? BART no
- Shall we add the good 2018 samples, especially from faeders?

# TO ADD/SPECIFY
- section on **Housing** of the birds
- specification of the heating table and video resolution (**Tomas**) 
- videos to accompany the protocols (vas deferens preparation, slide preparation, sperm collection and video recording)

# References